src/schedule/psk.rs

```diff
@@ -32,40 +34,71 @@ impl TryFrom<u8> for PSKType {
 
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
 pub struct ExternalPsk {
-    pub(crate) psk_id: Vec<u8>,
+    psk_id: Vec<u8>,
 }
 
 impl ExternalPsk {
+    pub fn new(psk_id: Vec<u8>) -> Self {
+        Self { psk_id }
+    }
     pub fn psk_id(&self) -> &[u8] {
         &self.psk_id
     }
 }
+
+#[cfg(test)]
+pub struct ExternalPskBundle {
+    secret: Secret,
+    nonce: Vec<u8>,
+    external_psk: ExternalPsk,
+}
+
+#[cfg(test)]
+impl ExternalPskBundle {
+    pub fn new(ciphersuite: &Ciphersuite, secret: Secret, psk_id: Vec<u8>) -> Self {
+        Self {
+            secret,
+            nonce: Ciphersuite::random_vec(ciphersuite.hash_length()),
+            external_psk: ExternalPsk { psk_id },
+        }
+    }
+    pub fn to_presharedkey_id(&self) -> PreSharedKeyID {
+        PreSharedKeyID {
+            psk_type: PSKType::External,
+            psk: Psk::External(self.external_psk.clone()),
+            psk_nonce: self.nonce.clone(),
+        }
+    }
+    pub fn secret(&self) -> &Secret {
+        &self.secret
+    }
+}
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
 pub struct ReinitPsk {
```

Comment:
Now that they're a bit more relevant, some comments would be nice on each of these structs.